src/components/NonEntitledSection/__tests__/NonEntitledSection.spec.js

```diff
@@ -3,28 +3,119 @@
+        title="test title"
+        subTitle="test subtitle"
+        description="test description"
+        links={[
+          {
+            id: 'test-link-1',
+            text: 'test link 1',
+            href: '#',
+          },
+          {
+            id: 'test-link-2',
+            text: 'test link 2',
+            href: '#',
+            icon,
+          },
+        ]}
+      />,
+      {
+        // DAP requires a landmark '<main>' in the DOM:
+        container: document.body.appendChild(main),
+      }
+    );
+
+    await expect(document.body).toHaveNoAxeViolations();
+    await expect(document.body).toHaveNoDAPViolations('NonEntitledSection');
+  });
+
+  test('should apply a title via `title`', () => {
+    const { queryByText } = render(
+      <NonEntitledSection title="test title" subTitle="test subtitle" />
+    );
+    expect(queryByText(/test title/i)).toBeVisible();
+  });
 
-  beforeEach(() => {
-    nonEntitledSection = shallow(
-      <NonEntitledSection {...noSubscriptionExample} />
+  test('should apply a subTitle via `subTitle`', () => {
+    const { queryByText } = render(
+      <NonEntitledSection title="test title" subTitle="test subtitle" />
     );
+    expect(queryByText(/test subtitle/i)).toBeVisible();
   });
 
-  describe('render', () => {
-    it('renders correctly', () => {
-      expect(nonEntitledSection).toMatchSnapshot();
-    });
+  test('should apply a description via `description`', () => {
+    const { queryByText } = render(
+      <NonEntitledSection
+        title="test title"
+        subTitle="test subtitle"
+        description="test description"
+      />
+    );
+    expect(queryByText(/test description/i)).toBeVisible();
+  });
 
-    it("renders the HTML of the node's subtree", () => {
-      expect(nonEntitledSection.render()).toMatchSnapshot();
-    });
+  test('should apply a link via `links` array of objects', () => {
+    const { queryByText } = render(
+      <NonEntitledSection
+        title="test title"
+        subTitle="test subtitle"
+        links={[
+          {
+            id: 'test-link',
+            text: 'test link',
+            href: '#',
+          },
+        ]}
+      />
+    );
+    expect(queryByText(/test link/i)).toBeVisible();
+  });
+
+  test('should apply a custom background image to style attribute via `backgroundImage`', () => {
+    const { container } = render(
+      <NonEntitledSection
+        title="test title"
+        subTitle="test subtitle"
+        backgroundImage="TEST-IMAGE"
+      />
+    );
+    expect(container.firstElementChild).toHaveAttribute(
+      'style',
+      `background-image: url(TEST-IMAGE);`
+    );
+  });
+
+  test('should apply a custom class name', () => {
+    const { container } = render(
+      <NonEntitledSection
+        title="test title"
+        subTitle="test subtitle"
+        className="custom-class"
+      />
+    );
+    expect(container.firstElementChild).toHaveClass('custom-class');
```

Comment:
I would have written this differently to rely less on the rendered structure from the testing library.

```suggestion
    expect(container.querySelector('.custom-class')).toBeInTheDocument();
```